cli/lib/search.go

```diff
@@ -193,3 +201,38 @@ func versionsFromSearchedLibrary(library *rpc.SearchedLibrary) []*semver.Version
 	sort.Sort(semver.List(res))
 	return res
 }
+
+// indexNeedsUpdating returns whether library_index.json need updating.
+// A positive duration string must be provided to calculate the time threshold
+// used to update the index.
+// Valid duration units are "ns", "us" (or "µs"), "ms", "s", "m", "h".
+// Use a duration of 0 to always update the index.
+func indexNeedsUpdating(duration string) bool {
+	// Library index path is constant (relative to the data directory).
+	// It does not depend on board manager URLs or any other configuration.
+	dataDir := configuration.Settings.GetString("directories.Data")
+	indexPath := paths.New(dataDir).Join("library_index.json")
+	// Verify the index file exists and we can read its fstat attrs.
+	if indexPath.NotExist() {
+		return true
+	}
+	info, err := indexPath.Stat()
+	if err != nil {
+		return true
+	}
+	// Sanity check the given threshold duration string.
+	now := time.Now()
+	modTimeThreshold, err := time.ParseDuration(duration)
+	if err != nil {
+		feedback.Error(tr("Invalid timeout: %s", err))
+		os.Exit(errorcodes.ErrBadArgument)
+	}
+	// The behavior of now.After(T) is confusing if T < 0 and MTIME in the future,
+	// and is probably not what the user intended. Disallow negative T and inform
+	// the user that positive thresholds are expected.
+	if modTimeThreshold < 0 {
+		feedback.Error(tr("Timeout must be non-negative: %dns (%s)", modTimeThreshold, duration))
+		os.Exit(errorcodes.ErrBadArgument)
+	}
```

Comment:
I would not bother too much with validating the timeout since it's a hardcoded constant.

```suggestion
```